### DISCUSSION TITLE
Linux (Ubuntu 20.04) - Debug and Publish self-contained experience

### DISCUSSION THREAD

**Discussion by Skrypt** [OrchardCMS/OrchardCore#7081] [Show and tell]
## Debug

First issue I encountered is related with .NET Core filewatchers : 

```
System.IO.IOException: The configured user limit (128) on the number of inotify instances has been reached.
```

To fix this issue under Ubuntu 20.04 I simply needed to do this command : 

```cmd
echo fs.inotify.max_user_instances=524288 | sudo tee -a /etc/sysctl.conf && sudo sysctl -p
```

## Publish (self-contained)

When publishing if I don't specify a runtime it publishes fine but when I specify I get errors. So for instance here I can't publish as self-contained under Ubuntu.

```
dotnet publish /repositories/OrchardCore/src/OrchardCore.Cms.Web --configuration release --self-contained --runtime linux-x64 --output ./build/release
```

Here are the errors :  

```
/repositories/OrchardCore/src/OrchardCore/OrchardCore.Application.Cms.Core.Targets/OrchardCore.Application.Cms.Core.Targets.csproj : error NU1605: Passage à une version antérieure du package détecté : System.Threading.ThreadPool de 4.3.0 à 4.0.10. Référencez le package directement à partir du projet pour sélectionner une version différente.  [/repositories/OrchardCore/src/OrchardCore.Cms.Web/OrchardCore.Cms.Web.csproj]
```

```
error NU1605:  OrchardCore.Cms.Web -> OrchardCore.Application.Cms.Targets -> TheTheme -> OrchardCore.Menu -> OrchardCore.ContentManagement.GraphQL -> OrchardCore.Apis.GraphQL.Abstractions -> GraphQL 2.4.0 -> System.Reactive.Core 3.1.1 -> System.Threading.ThreadPool (>= 4.0.10)
```

See related issue : 

https://github.com/dotnet/installer/issues/81

They seem to say it should be resolved by using `<TargetLatestRuntimePatch>true</TargetLatestRuntimePatch>` in the .csproj file. I need to test this more but so far by adding this in OrchardCore.Cms.Web only did not work.

**Comment by sebastienros** [comment]
I wonder if the FileSystemWatcher issue would repro with a standard aspnet app.
Also we might want to check the switch to disable them.

**Comment by Skrypt** [comment]
Not sure why this is assigned to 1.0. I just wanted to document my experience with Ubuntu desktop here. I'm not sure this is worth creating a documentation for this as this information could change over time.

The publish self-contained not working though is something that we can't really fix if it's coming from .NET Core issues.

**Comment by Skrypt** [comment]
I tried this back this week and it worked "Publish self contained". Under Ubuntu 20 still.
